src/hotspot/share/runtime/basicLock.cpp

```diff
@@ -31,7 +31,7 @@
 
 void BasicLock::print_on(outputStream* st, oop owner) const {
   st->print("monitor");
-  if (LockingMode == LM_LIGHTWEIGHT) {
+  if (UseObjectMonitorTable) {
```

Comment:
This was a pre-existing bug. It should be:
```c++
  if (UseObjectMonitorTable) {
    ObjectMonitor* mon = object_monitor_cache();
    if (mon != nullptr) {
      mon->print_on(st);
    }
  } else if (LockingMode == LM_LEGACY) {
```
I guess no testing ever printed a BasicLock with LM_MONITOR.